simd/unit_tests/include/TestSIMD_LoadStore.hpp

```diff
@@ -0,0 +1,237 @@
+
+  host_test_simd_load(expected, expected,
+                      Kokkos::Experimental::simd_flag_default);
+  host_test_simd_load(expected, expected,
+                      Kokkos::Experimental::simd_flag_aligned);
+  host_test_simd_load(expected, expected_masked, mask,
+                      Kokkos::Experimental::simd_flag_default);
+  host_test_simd_load(expected, expected_masked, mask,
+                      Kokkos::Experimental::simd_flag_aligned);
+}
+
+template <typename Abi, typename DataType>
+inline void host_check_loadstore() {
+  if constexpr (is_type_v<Kokkos::Experimental::basic_simd<DataType, Abi>>) {
+    host_test_simd_loadstore<Abi, DataType>();
+  }
+}
+
+template <typename Abi, typename... DataTypes>
+inline void host_check_loadstore_all_types(
+    Kokkos::Experimental::Impl::data_types<DataTypes...>) {
+  (host_check_loadstore<Abi, DataTypes>(), ...);
+}
+
+template <typename... Abis>
+inline void host_check_loadstore_all_abis(
+    Kokkos::Experimental::Impl::abi_set<Abis...>) {
+  using DataTypes = Kokkos::Experimental::Impl::data_type_set;
+  (host_check_loadstore_all_types<Abis>(DataTypes()), ...);
+}
+
+template <typename SimdType, typename... Args>
+KOKKOS_INLINE_FUNCTION void device_test_simd_load(SimdType const& init,
+                                                  SimdType const& expected,
+                                                  Args... args) {
+  constexpr size_t alignment =
+      SimdType::size() * sizeof(typename SimdType::value_type);
+
+  alignas(alignment) typename SimdType::value_type arr[SimdType::size()] = {0};
+  simd_unchecked_store(init, arr, Kokkos::Experimental::simd_flag_default);
+
+  SimdType result;
+
+  if constexpr (sizeof...(args) > 1) {
+    result = simd_partial_load(arr, args...);
+  } else {
+    if constexpr (std::is_same_v<
+                      Kokkos::Experimental::simd_abi::Impl::host_fixed_native,
+                      typename SimdType::abi_type>) {
+      result = Kokkos::Experimental::simd_unchecked_load(arr, args...);
+    } else {
+      result =
+          Kokkos::Experimental::simd_unchecked_load<SimdType>(arr, args...);
+    }
+  }
+
+  for (size_t i = 0; i < SimdType::size(); ++i) {
+    EXPECT_EQ(result[i], expected[i]);
+  }
+}
+
+template <typename SimdType, typename... Args>
+KOKKOS_INLINE_FUNCTION void device_test_simd_store(SimdType const& init,
+                                                   SimdType const& expected,
+                                                   Args... args) {
+  constexpr size_t alignment =
+      SimdType::size() * sizeof(typename SimdType::value_type);
+
+  alignas(alignment) typename SimdType::value_type arr[SimdType::size()] = {0};
+
+  if constexpr (sizeof...(args) > 1) {
+    simd_partial_store(init, arr, args...);
+  } else {
+    simd_unchecked_store(init, arr, args...);
+  }
+
+  for (size_t i = 0; i < SimdType::size(); ++i) {
+    EXPECT_EQ(arr[i], expected[i]);
+  }
```

Comment:
```suggestion
  kokkos_checker checker;
  for (size_t i = 0; i < SimdType::size(); ++i) {
    checker.equality(arr[i], expected[i]);
  }
```
Same, `EXPECT_EQ` will not compile on device, and `device_check_equality` cannot be used as `arr` is an array

Reply:
Also fixed.